include/VecCore/Backend/Implementation.h

```diff
@@ -284,6 +285,21 @@ VECCORE_FORCE_INLINE VECCORE_ATT_HOST_DEVICE constexpr Bool_s EarlyReturnAllowed
 #endif
 }
 
+// An extension of EarlyReturnAllowed(), and new name is a reminder that it can be used for other purposes as well
+template <typename T, size_t simdSize = 1>
+VECCORE_FORCE_INLINE
+VECCORE_ATT_HOST_DEVICE
+constexpr Bool_s SIMDsizeUpTo()
```

Comment:
> Yes and on CUDA, at the moment the vector size is always one and we are claiming that making the thread going out-of-sync (which any early return would do) is more costly than continuing ...

Sure, but your logic is a bit backwards. It's very counter intuitive that `SIMDsizeUpTo<double, 1>()` would ever return false. It would be better to use `SIMDSizeAtLeast<T, N>()` instead, where it returns false unless the width is at least N. Then you can use the same code for SIMD and CUDA without any special case, by asking for N=2. The fact that `EarlyReturnAllowed() == false` for CUDA will in any case make the result of `EarlyReturnAllowed() && SIMDSizeAtLeast<T,N>()` return always false even for N=1 as you wanted.

Of course, in case you need to test for small N, you can use `EarlyReturnAllowed() && !SIMDSizeAtLeast<T,N>()`, which still returns false for CUDA and true for scalar and short SIMD.

As for long lines, you can always break things into two if statements:
```cpp
if (vecCore::EarlyReturnAllowed() && vecCore::VectorSize<Double_v>() < N)
    if (vecCore::MaskFull(done)) return;
```
or, with using namespace vecCore:
```cpp
using namespace vecCore;
if (EarlyReturnAllowed() && VectorSize<Double_v>() < N)
    if (MaskFull(done)) return;
```
I actually think this last version, which doesn't require anything extra in VecCore, is the most clearly legible and easy to understand, and there is no performance penalty, because the first `if` is evaluated at compile time.